IdentityCore/src/network/request/MSIDAADAuthorityMetadataRequest.m

```diff
@@ -59,4 +59,20 @@ - (instancetype)initWithEndpoint:(NSURL *)endpoint
     return self;
 }
 
+- (NSCachedURLResponse *)cachedResponse
```

Comment:
nit: code in MSIDAADAuthorityMetadataRequest and MSIDOpenIdConfigurationInfoRequest looks duplicate. Possible to have it in one place...?

Reply:
Addressed. Added a boolean flag shouldCacheResponse with default value NO to base MSIDHttpRequest and have subclasses override that property if they want nsurlcache enabled.